### PR TITLE
fix: add first version of the complete architecture running successfully

### PR DESCRIPTION
**Pull request by Kleiber**
**Dependencies**

``` bash
$ go get -u github.com/stretchr/testify
$ go get -u go.uber.org/zap
$ go get -u github.com/gorilla/mux
$ go get -u github.com/gorilla/handlers
$ go get -u github.com/go-sql-driver/mysql
```

**Scripts**
On MySQL Workbench connect with the user `root` and execute the following scripts

_script:_ `dbtest.sql`
``` bash
$  cd $GOPATH/src/github.com/QOSQOs/UNIVeasier/internal/scripts/databases/
```
_script:_ `test.sql`
``` bash
$   cd $GOPATH/src/github.com/QOSQOs/UNIVeasier/internal/scripts/procedures/
```

**Build the project binary**

``` bash
$ cd $GOPATH/src/github.com/QOSQOs/UNIVeasier/
$ go install $GOPATH/src/github.com/QOSQOs/UNIVeasier/cmd/
```

**Execute the project binary**
``` bash
$ cd $GOPATH/bin/
$ ./cmd.exe
```

**Run unit test**

Go to the folder where the file containing the tests is located ans run:

```bash
$ cd $GOPATH/src/github.com/QOSQOs/UNIVeasier/pkg/model/test/ 
$ go test -v
```

To run all unit tests of the project:

``` bash
$ cd $GOPATH/src/github.com/QOSQOs/UNIVeasier/
$ go test ./...
```

**Test the REST API**

``` bash
$ curl -s -X GET http://localhost:8000/test
$ curl -s -X GET http://localhost:8000/test/{name}
```

### CODE REVIEW

File: pkg/api/server.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	s.RoutesTest()
+}
+
+func (s *Server) Initialize(filename string) error {
+	err := s.initLogger()
+	if err != nil {
+		return err
+	}
+	common.Log = s.Log
+	common.Global = "variable"
```
Comment (darwinTtito) [inline pkg/api/server.go]: delete this unused variable.

File: pkg/api/server.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+	return nil
+}
+
+func (s *Server) Run() {
+	headers := handlers.AllowedHeaders([]string{"X-Requested-With", "Content-Type", "Authorization"})
+	methods := handlers.AllowedMethods([]string{"GET", "POST", "PUT", "DELETE"})
+	origins := handlers.AllowedOrigins([]string{"*"})
+
+	s.Log.Info("Server listening on http://localhost:8000")
```
Comment (darwinTtito) [inline pkg/api/server.go]: Uses the info included in the DBConfig file (host, port, etc) to build this log message.

File: pkg/api/server.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+	err := http.ListenAndServe(":8000", handlers.CORS(headers, methods, origins)(s.Router))
```
Comment (darwinTtito) [inline pkg/api/server.go]: similar that the previous comment

File: pkg/model/test/testModel.go [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+		return errors.New("Invalid test age")
+	}
+
+	fmt.Println("3")
+
+	ok = common.Emails.MatchString(t.Email)
+	if len(t.Email) == 0 || !ok {
+		return errors.New("Invalid test email")
+	}
+	fmt.Println("4")
```
Comment (darwinTtito) [inline pkg/model/test/testModel.go]: remove the unnecessary Println instruction.

**Review by darwinTtito — CHANGES_REQUESTED** [review]
Read the comments.

File: pkg/api/server.go (hunk 1 above)
Comment (Kleiber) [reply pkg/api/server.go]: Done!

File: pkg/model/test/testModel.go (hunk 4 above)
Comment (Kleiber) [reply pkg/model/test/testModel.go]: Done!

File: pkg/api/server.go (hunk 3 above)
Comment (Kleiber) [reply pkg/api/server.go]: Done!

File: pkg/api/server.go (hunk 2 above)
Comment (Kleiber) [reply pkg/api/server.go]: Done!

**Review by darwinTtito — APPROVED** [review]

**Pull request merged**